### PR TITLE
Document Getting Started with Prometheus and Grafana

### LINKED ISSUE #1340: [Metrics] Document Getting Started with Prometheus and Grafana
It would be good to have a getting started guide to setup Metrics exporter for Prometheus and demonstrate end to end metrics flow. 

Good reference - 
https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/docs/metrics/getting-started-prometheus-grafana

Depends on these PRs get merged.
- #1331
- #1332

### PR DESCRIPTION
**Pull request by esigo**
Fixes #1340 (issue)

## Changes

adapted from [dotnet repo](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/docs/metrics/getting-started-prometheus-grafana/README.md).

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

File: examples/prometheus/README.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+It is highly recommended to go over the [ostream-metrics](../metrics_simple/README.md)
+doc before following along this document.
+
+Create a new console application and run it:
+
+```sh
+bazel run //examples/prometheus:prometheus_example
+```
+
+Now, we are going to make some small tweaks to the example in the
```
Comment (lalitb) [inline examples/prometheus/README.md:23]: Should we be making these changes in the example before running it?

Comment (lalitb) [reply examples/prometheus/README.md:23]: Sorry if I am missing something. Why do we need to modify ostream-metrics to export to Prometheus Exporter - the above step of running prometheus_example should already do the metrics export?

Comment (esigo) [reply examples/prometheus/README.md]: thanks, it is not needed to do anything extra here. I changed the wording to make it clear.

File: examples/prometheus/README.md [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+## Collect metrics using Prometheus
+
+Follow the [first steps](https://prometheus.io/docs/introduction/first_steps/)
+to download the [latest release](https://prometheus.io/download/) of Prometheus.
+It is also possible to use `prom/prometheus` docker image.
+
+### Configuration
+
+After finished downloading, extract it to a local location that's easy to
```
Comment (lalitb) [inline examples/prometheus/README.md:95]:
```suggestion
After downloading, extract it to a local location that's easy to
```

**Review by lalitb — APPROVED** [review]
LGTM. Thanks

File: examples/prometheus/README.md [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+```cpp
+std::unique_ptr<metrics_sdk::MetricExporter> exporter{
+    new metrics_exporter::PrometheusExporter(opts)};
+```
+
+OpenTelemetry `PrometheusExporter` will export
+data via the endpoint defined by
+`metrics_exporter::PrometheusExporterOptions::url`,
+which is `http://localhost:8080/` by default.
```
Comment (reyang) [inline examples/prometheus/README.md:42]: Not directly related to this PR, consider changing the default port following https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#prometheus-exporter (note that the spec is still Experimental).

File: examples/prometheus/README.md [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+
+```mermaid
+graph TD
+
+subgraph Prometheus
+  PrometheusScraper
+  PrometheusDatabase
+end
+
+PrometheusExporter["PrometheusExporter<br/>(listening at #quot;http://localhost:8080/#quot;)"] -->|HTTP GET| PrometheusScraper{{"Prometheus scraper<br/>(polling #quot;http://localhost:8080/metrics#quot; every 10 seconds)"}}
```
Comment (reyang) [inline examples/prometheus/README.md:197]:
```suggestion
PrometheusExporter["PrometheusExporter<br/>(listening at #quot;http://localhost:8080/#quot;)"] -->|HTTP GET| PrometheusScraper{{"Prometheus scraper<br/>(polling #quot;http://localhost:8080/metrics#quot; every 5 seconds)"}}
```

**Review by reyang — APPROVED** [review]

File: examples/prometheus/README.md (hunk 3 above)
Comment (esigo) [reply examples/prometheus/README.md]: thanks, I fixed the port.

**Pull request merged**